Library/Homebrew/tap.rb

```diff
@@ -143,14 +143,46 @@ def official?
     user == "Homebrew"
   end
 
+  # Set the cached Git configuration for whether or not this tap is private.
+  def set_private_config
+    return if !git? || !Utils.git_available?
+
+    path.cd do
+      private_tap = begin
+        if custom_remote?
+          true
+        else
+          GitHub.private_repo?(user, "homebrew-#{repo}")
+        end
+      rescue GitHub::HTTPNotFoundError
+        true
+      rescue GitHub::Error
+        false
+      end
+
+      system("git", "config", "--replace-all", "remote.origin.private", private_tap.to_s)
+    end
+  end
+
+  # @private
+  def private_config
+    if git? && Utils.git_available?
+      path.cd do
+        private_config = Utils.popen_read("git", "config", "--get", "remote.origin.private").chomp
+        if private_config.to_s.empty?
+          set_private_config
+          private_config = Utils.popen_read("git", "config", "--get", "remote.origin.private").chomp
+        end
+        private_config
+      end
+    end
+  end
+
   # True if the remote of this {Tap} is a private repository.
   def private?
-    return true if custom_remote?
-    GitHub.private_repo?(user, "homebrew-#{repo}")
-  rescue GitHub::HTTPNotFoundError
-    true
-  rescue GitHub::Error
-    false
+    @private ||= begin
+      private_config == "true"
```

Comment:
For normal tap, i.e. `@private` is false. The cache won't work, as result, it will introduce unnecessary IO.